mitodl/mitxpro
# Added support for External Programs in product page & added course carousel

#### Pre-Flight checklist




#### What are the relevant tickets?
It is related to #2045 (Will close it manually)

#### What's this PR do?
- Adds support for carousel view for both CMS/Product Page for External Programs

#### How should this be manually tested?
- Visit `/cms` and create an External Program Page
- In the above created External Program add a child page `Course Caraousal/Courses-In-Program` page
- Add course pages to the content of the Carousel page you're creating
- Check the `Override Content` checkbox on the carousel page you're creating
- Now load the program details/product page and make sure you are able to see the associated courses in ExternalProgram

#### Where should the reviewer start?
Creating/Updating External Program Page in cms

#### Screenshots (if appropriate)
**External Program Page list CMS**
<img width="1236" alt="Screenshot 2021-01-11 at 5 04 51 PM" src="https://user-images.githubusercontent.com/34372316/104180704-633bed80-542f-11eb-8f60-3cb8004a8930.png">
**Available child pages in External Program Page CMS**
<img width="1221" alt="Screenshot 2021-01-11 at 5 09 08 PM" src="https://user-images.githubusercontent.com/34372316/104180953-ccbbfc00-542f-11eb-99c8-e76a66a24fd1.png">

**External Program Page Add Carousel**
<img width="1196" alt="Screenshot 2021-01-11 at 5 05 40 PM" src="https://user-images.githubusercontent.com/34372316/104180655-4acbd300-542f-11eb-883b-8d5d8b235d0e.png">

**Caourses carousel in Externa Product page **
<img width="1024" alt="Screenshot 2021-01-11 at 5 05 27 PM" src="https://user-images.githubusercontent.com/34372316/104180703-62a35700-542f-11eb-85cd-28c22dc63762.png">
